sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -948,6 +975,14 @@ public void setTimestampPolicy(String timestampPolicy) {
         public void setConsumerPollingTimeout(Long consumerPollingTimeout) {
           this.consumerPollingTimeout = consumerPollingTimeout;
         }
+
+        public void setRedistributeNumShards(Long redistributeNumShards) {
```

Comment:
Since it is boxed, you can make it `@Nullable` and that way you can check whether it has been set or not. It is fine to rely on defaults to zero also, but then make it unboxed (if you can - I don't know how autovalue works with that)

Reply:
Left it as defaulting to zero.